pkgs/by-name/co/cotton/package.nix

```diff
@@ -1,21 +1,30 @@
 {
   lib,
+  fetchpatch2,
   rustPlatform,
   fetchFromGitHub,
 }:
 
 rustPlatform.buildRustPackage {
   pname = "cotton";
-  version = "0-unstable-2023-09-13";
+  version = "0-unstable-2025-08-24";
 
   src = fetchFromGitHub {
     owner = "danielhuang";
     repo = "cotton";
-    rev = "df9d79a4b0bc4b140e87ddd7795924a93775a864";
-    sha256 = "sha256-ZMQaVMH8cuOb4PQ19g0pAFAMwP8bR60+eWFhiXk1bYE=";
+    rev = "43eb56f8d0228b63f26a48491980d4e91e965b8e";
+    hash = "sha256-GF5RtQmzPCDp5H1EW5rl+GDr2/xKA9zrjOeYyCdBtmI=";
   };
 
-  cargoHash = "sha256-y162tjxPGZQiYBJxFk4QN9ZqSH8jrqa5Y961Sx2zrRs=";
+  cargoPatches = [
+    (fetchpatch2 {
+      name = "Fix-CVE-2025-62518";
+      url = "https://github.com/danielhuang/cotton/commit/586fb861e9cef540dd3973210d82923436b8bc4b.patch?full_index=1";
```

Comment:
Upstream PR was closed and PR branch deleted, so this URL might disappear soon (making this review blocking until the patch has been vendored or cotton is dropped)